notification-portlet-webapp/src/main/webapp/scripts/modal.js

```diff
@@ -0,0 +1,129 @@
+Vue.component("modal", {
+  template: "#modal-template",
+  props: {
+    id: String
+  }
+})
+
+// start app
+new Vue({
+  el: "#notifications-vue-app",
+  data: {
+    items: [{ show: false }],
```

Comment:
why does items start with an item?
doesn't the UI start with no items, then populate them via API call?

Reply:
I started with an item so that the UI didn't break. I use the `show` value to determine to show a modal or not. If there are no items, then I think the modal will always show up

Reply:
> If there are no items, then I think the modal will always show up

Does that also happens when the API returns zero notifications?

Reply:
With the way the code is right now, if there are zero notifications, the modal will NOT show up.